JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreen.kt

```diff
@@ -121,103 +139,160 @@ fun HomeScreen(
 
-        // Get the text to show on the message from resources
-        val errorMessageText: String = stringResource(errorMessage.messageId)
-        val retryMessageText = stringResource(id = R.string.retry)
+                // Get the text to show on the message from resources
+                val errorMessageText: String = stringResource(errorMessage.messageId)
+                val retryMessageText = stringResource(id = R.string.retry)
 
-        // If onRefreshPosts or onErrorDismiss change while the LaunchedEffect is running,
-        // don't restart the effect and use the latest lambda values.
-        val onRefreshPostsState by rememberUpdatedState(onRefreshPosts)
-        val onErrorDismissState by rememberUpdatedState(onErrorDismiss)
+                // If onRefreshPosts or onErrorDismiss change while the LaunchedEffect is running,
+                // don't restart the effect and use the latest lambda values.
+                val onRefreshPostsState by rememberUpdatedState(onRefreshPosts)
+                val onErrorDismissState by rememberUpdatedState(onErrorDismiss)
 
-        // Effect running in a coroutine that displays the Snackbar on the screen
-        // If there's a change to errorMessageText, retryMessageText or scaffoldState,
-        // the previous effect will be cancelled and a new one will start with the new values
-        LaunchedEffect(errorMessageText, retryMessageText, scaffoldState) {
-            val snackbarResult = scaffoldState.snackbarHostState.showSnackbar(
-                message = errorMessageText,
-                actionLabel = retryMessageText
-            )
-            if (snackbarResult == SnackbarResult.ActionPerformed) {
-                onRefreshPostsState()
+                // Effect running in a coroutine that displays the Snackbar on the screen
+                // If there's a change to errorMessageText, retryMessageText or scaffoldState,
+                // the previous effect will be cancelled and a new one will start with the new values
+                LaunchedEffect(errorMessageText, retryMessageText, scaffoldState) {
+                    val snackbarResult = scaffoldState.snackbarHostState.showSnackbar(
+                        message = errorMessageText,
+                        actionLabel = retryMessageText
+                    )
+                    if (snackbarResult == SnackbarResult.ActionPerformed) {
+                        onRefreshPostsState()
+                    }
+                    // Once the message is displayed and dismissed, notify the ViewModel
+                    onErrorDismissState(errorMessage.id)
+                }
+            }
+        } else {
+            val selectedPost = uiState.posts.find { it.id == uiState.selectedPostId }
+
+            if (selectedPost != null) {
+                val lazyListState = detailLazyListStates.getValue(selectedPost.id)
+
+                ArticleScreen(
```

Comment:
Weird to have the `ArticleScreen` call on its own here. Is this full screen and for Compat Windows? Cannot follow the logic in here. More comments are needed.

But if this is for mobile phones, let's see if it's possible to make a navigation call

Reply:
Oh! It might be the case we don't call `navigate(ARTICLE_SCREEN)` anymore, is that the case? Then we wouldn't be able to use the `ArticleViewModel`. This looks ... complicated

Reply:
Oh! It might be the case we don't call `navigate(ARTICLE_SCREEN)` anymore, is that the case? Then we wouldn't be able to use the `ArticleViewModel`. This looks ... complicated

Reply:
Right, I'm avoiding doing any sort of `navigate` as a side-effect of a resize, which implies that we need to show the `ArticleScreen` (in mostly its entirety, with the bottom bar, etc.) at this navigation route.